### PR TITLE
charts: split properties for webhook and manager

### LINKED ISSUE #1259: chart: expose the `env` property for the kubefed components in `values.yaml`
**What would you like to be added**:
We should expose the `env` properties of the deployment specs for the kubefed pods in case we need to add ENV variables such as HTTP_NOPROXY, ...

**Why is this needed**:

/kind feature

### PR DESCRIPTION
**Pull request by hectorj2f**
**What this PR does / why we need it**:

We should split the webhook properties from the manager properties.
Also we should expose the env properties for the chart as well.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

Closes #1259 

**Special notes for your reviewer**:

### CODE REVIEW

File: charts/kubefed/values.yaml [hunk 1]
```diff
@@ -6,19 +6,36 @@
...
+    logLevel: 2
+    env: {}
+    resources:
+      limits:
+        cpu: 500m
+        memory: 512Mi
+      requests:
+        cpu: 100m
+        memory: 64Mi
+  webhook:
```
Comment (RainbowMango) [inline charts/kubefed/values.yaml:24]: Why we put `webhook`'s properties under `controllermanager`?

File: charts/kubefed/values.yaml [hunk 2]
```diff
@@ -6,19 +6,36 @@
 ##
 controllermanager:
```
Comment (RainbowMango) [inline charts/kubefed/values.yaml:7]: (Not sure if it works.) Just a guess, would it be better to rename `controllermanager` to other common names such as `component`?

**Comment by RainbowMango** [comment]
Just a nit, otherwise LGTM.

File: charts/kubefed/values.yaml [hunk 3]
```diff
@@ -6,19 +6,36 @@
...
-  imagePullPolicy: IfNotPresent
-  logLevel: 2
-  resources:
-    limits:
-      cpu: 500m
-      memory: 512Mi
-    requests:
-      cpu: 100m
-      memory: 64Mi
+  manager:
```
Comment (irfanurrehman) [inline charts/kubefed/values.yaml:9]: suggestion: I would rather name this `controller`.

File: charts/kubefed/values.yaml [hunk 4]
```diff
@@ -6,19 +6,36 @@
...
+    env: {}
+    resources:
+      limits:
+        cpu: 100m
+        memory: 256Mi
+      requests:
+        cpu: 100m
+        memory: 64Mi
+
   clusterAvailableDelay:
```
Comment (irfanurrehman) [inline charts/kubefed/values.yaml:39]: Aren't these properties also applicable only to `controller`?

File: charts/kubefed/values.yaml (hunk 1 above)
Comment (hectorj2f) [reply charts/kubefed/values.yaml:24]: They are part of the chart components named controller manager.

File: charts/kubefed/values.yaml (hunk 2 above)
Comment (hectorj2f) [reply charts/kubefed/values.yaml:7]: That might imply to change the chart name. I don't know if we really want to do that now.

File: charts/kubefed/values.yaml (hunk 3 above)
Comment (hectorj2f) [reply charts/kubefed/values.yaml:9]: Yes, we could use `controller` too.

File: charts/kubefed/values.yaml (hunk 1 above)
Comment (irfanurrehman) [reply charts/kubefed/values.yaml:24]: The webhook can actually be running external to the application, however this was for the sake of simplicity in the first place, probably not to define many new additional settings.

File: charts/kubefed/values.yaml (hunk 4 above)
Comment (hectorj2f) [reply charts/kubefed/values.yaml:39]: These are part of `controllermanager` main chart, perhaps I can move them up to easily identify the scope for manager` and `webhook`.

File: charts/kubefed/values.yaml (hunk 1 above)
Comment (hectorj2f) [reply charts/kubefed/values.yaml:24]: I didn't want to reorganize the whole chart before we get this potential PR with Helmv3 support. But I could do it if you want it.

Comment (irfanurrehman) [reply charts/kubefed/values.yaml:24]: I did not imply to move it out. I am ok until there is a need for it. :) (YAGNI)

File: charts/kubefed/values.yaml (hunk 4 above)
Comment (hectorj2f) [reply charts/kubefed/values.yaml]: I moved them up so it is more clear the purpose of the properties manager and webhook. To me, properties such as `clusterAvailableDelay` are defining a more generic behavior to the whole kubefed control-plane.

File: charts/kubefed/charts/controllermanager/templates/deployments.yaml [hunk 5]
```diff
@@ -22,10 +22,17 @@ spec:
       containers:
       - command:
         - /hyperfed/controller-manager
-        - "--v={{ .Values.logLevel }}"
-        image: "{{ .Values.repository }}/{{ .Values.image }}:{{ .Values.tag }}"
-        imagePullPolicy: "{{ .Values.imagePullPolicy }}"
+        - "--v={{ .Values.manager.logLevel }}"
```
Comment (irfanurrehman) [inline charts/kubefed/charts/controllermanager/templates/deployments.yaml:25]: Wouldn't you need to change the to `controller`? As you have updated in `values.yaml` below.

Comment (hectorj2f) [reply charts/kubefed/charts/controllermanager/templates/deployments.yaml]: yes, I did now pushed the changes.

**Comment by RainbowMango** [comment]
LGTM. I'd like to give a chance to @irfanurrehman for final approval.

**Comment by irfanurrehman** [comment]
Thanks for doing this @hectorj2f.
/lgtm
/approve

**Comment by hectorj2f** [comment]
Thanks for the reviews.

**Pull request merged**